app/src/main/java/com/kabouzeid/gramophone/ui/fragments/player/PlayerAlbumCoverFragment.java

```diff
@@ -171,6 +217,52 @@ public void setCallbacks(Callbacks listener) {
         callbacks = listener;
     }
 
+    @Override
+    public void onUpdateProgressViews(int progress, int total) {
+        if (lyricsLayout == null || lyricsLine1 == null || lyricsLine2 == null) return;
+
+        if (!PreferenceUtil.getInstance(getActivity()).synchronizedLyricsShow() || lyrics == null || !lyrics.isSynchronized() || !lyrics.isValid()) {
```

Comment:
This is the same logic as in `setLyrics`, so I think it could be extracted into a method.